zipkin-storage/cassandra3/src/main/resources/cassandra3-schema.cql

```diff
@@ -1,88 +1,100 @@
 
-CREATE TYPE IF NOT EXISTS zipkin3.trace_id (
-    high    bigint,
-    low     bigint
-);
-
-CREATE TYPE IF NOT EXISTS zipkin3.endpoint (
-    service_name text,
+CREATE TYPE IF NOT EXISTS zipkin2_cassandra3.endpoint (
+    service      text,
     ipv4         inet,
     ipv6         inet,
-    port         smallint,
+    port         int,
 );
 
-CREATE TYPE IF NOT EXISTS zipkin3.annotation (
+CREATE TYPE IF NOT EXISTS zipkin2_cassandra3.annotation (
     ts bigint,
     v  text,
-    ep frozen<endpoint>
-);
-
-CREATE TYPE IF NOT EXISTS zipkin3.binary_annotation (
-    k  text,
-    v  blob,
-    t  text,
-    ep frozen<endpoint>
 );
 
-CREATE TABLE IF NOT EXISTS zipkin3.traces (
-    trace_id            frozen<trace_id>,
+CREATE TABLE IF NOT EXISTS zipkin2_cassandra3.span (
+    trace_id            text,
     ts_uuid             timeuuid,
-    id                  bigint,
+    id                  text,
     ts                  bigint,
-    span_name           text,
-    parent_id           bigint,
+    span                text,
+    parent_id           text,
     duration            bigint,
+    l_ep                Endpoint,
+    l_service           text,
+    r_ep                Endpoint,
     annotations         list<frozen<annotation>>,
-    binary_annotations  list<frozen<binary_annotation>>,
-    all_annotations     text, //-- can't do SASI on set<text>: comma-joined until CASSANDRA-11182
+    tags                map<text,text>,
+    shared              boolean,
+    annotation_query    text, //-- can't do SASI on set<text>: comma-joined until CASSANDRA-11182
     PRIMARY KEY (trace_id, ts_uuid, id)
 )
     WITH CLUSTERING ORDER BY (ts_uuid DESC)
     AND compaction = {'class': 'org.apache.cassandra.db.compaction.TimeWindowCompactionStrategy'}
-    AND default_time_to_live =  604800;
+    AND default_time_to_live =  604800
+    AND gc_grace_seconds = 3600
+    AND read_repair_chance = 0
+    AND speculative_retry = '95percentile'
+    AND comment = 'Primary table for holding trace data. annotation_query column is a comma separated string (instead of a set) to permit SASI (CASSANDRA-11182)';
 
 
-CREATE TABLE IF NOT EXISTS zipkin3.trace_by_service_span (
-    service_name  text,             //-- service name
-    span_name     text,             //-- span name, or blank for queries without span name
+CREATE TABLE IF NOT EXISTS zipkin2_cassandra3.trace_by_service_span (
+    service       text,             //-- service name
+    span          text,             //-- span name, or blank for queries without span name
     bucket        int,              //-- time bucket, calculated as ts/interval (in microseconds), for some pre-configured interval like 1 day.
     ts            timeuuid,         //-- start timestamp of the span, truncated to millisecond precision
-    trace_id      frozen<trace_id>, //-- trace ID
+    trace_id      text,             //-- trace ID
     duration      bigint,           //-- span duration, in microseconds
```

Comment:
@llinder @fedj 
i want to change duration to be stored as tens of milliseconds (or in hundredths of seconds).
this will make the SASI on this column contains fewer distinct values, improving write performance.
and i can't see users being so discerning about searches they perform in the UI.

duration would be rounded up, so the result list would not skip any results. and duration in the span table would remain accurate to microseconds.

Reply:
Sounds like a good optimization to me.

Reply:
cc @openzipkin/cassandra

Reply:
added as an extra commit. can be undone if any objections/concerns.